apps/mobile/src/utils/analytics.ts

```diff
@@ -16,15 +16,43 @@ const segmentClient = createClient({
 
 segmentClient.add({ plugin: new AppLifecycleEventPlugin() });
 
-const leatherAnalyticsClient = configureAnalyticsClient<SegmentClient>({
+const analyticsClient = configureAnalyticsClient({
   client: segmentClient,
   defaultProperties: {
     platform: 'mobile',
   },
 });
 
-export let analytics: ReturnType<typeof configureAnalyticsClient<SegmentClient>> | undefined =
-  leatherAnalyticsClient;
+let analyticsEnabled = selectAnalyticsPreference(store.getState()) === 'consent-given';
+
+function setupPreferenceSubscription() {
+  return store.subscribe(() => {
+    const analyticsPreference = selectAnalyticsPreference(store.getState());
+
+    if (analyticsPreference === 'rejects-tracking') {
+      void analytics.track('user_setting_updated', { analytics: 'rejects-tracking' });
+    }
+
+    analyticsEnabled = analyticsPreference === 'consent-given';
+  });
+}
+
+setupPreferenceSubscription();
+
+export const analytics = new Proxy(analyticsClient, {
+  get(target, prop, receiver) {
+    const value = Reflect.get(target, prop, receiver);
+
+    if (typeof value === 'function') {
+      return (...args: unknown[]) => {
+        if (!analyticsEnabled) return;
+        return value.apply(target, args);
+      };
+    }
+
+    return value;
+  },
+});
```

Comment:
So the proxy just returns undefined to any fn call if analytics isn't enabled? 

I wonder if this is necessary, as the client already works in such a way that you can call all the methods but nothing happens if disabled (as far as I understand).

Reply:
Correct, just noop-ing all possible (unknown) methods.
The assumption here is that the client is [our own client adapter](https://github.com/leather-io/mono/blob/69566995267061d4661e6094c04d21b9a732deaf/packages/analytics/src/client.ts) not Segment, especially now that we might use something else.

In hindsight though, I'd love our `analytics` module to have `enabled` or any sort off a an on/off switch for this, so other apps could use it too. Something to look into.